### DISCUSSION TITLE
No admin account on my selfhosted instance

### DISCUSSION THREAD

**Discussion by elythh** [outline/outline#3468] [Self Hosting] [answered]
Hello, I've been selfhosting Outline for about a year now and I wanted to move my current instance to a new one. I realized that I wasn't able to export all my collections because my current and only account doesn't have admin rights. On the new one, I followed the exact same installation process and my account is Admin. 

How can I promote my current user to admin ? From the API I need to already be one to promote a user.

Thank you for your help !

EDIT: I'm using Slack for authentication and both instance are using the same Slack account

**Comment by tommoor** [comment, marked as answer]
You would need to change the `user.isAdmin` column in the database directly in this scenario

**Reply by elythh** [reply]
Thank you for the quick answer. Could you help me change it ? My database is running in a docker container. I am able to access it using `docker exec -tiu postgres outline-postgres psql` but from that point I'm not familiar with the commands to edit the right table.

EDIT: I located the table but I can't find how to edit the `False` value to `True`.
EDIT2: Nevermind, I had to quote "isAdmin" when using UPDATE. Thank you for the help !

**Reply by imedia3** [reply]
You might post your final steps with found solution, as you expected to receive from others

**Reply by elythh** [reply]
I felt like tommor's response and mine were enough but I will try to explain each step here. 

* If you are running docker, log into your postgre container ( the commands is in my above comment) 
* Find the `user` table
* Update the right user changing `isAdmin` column from `False` to `True`

I don't have the excact postgre syntax in head unfortunately

**Reply by techguydave** [reply]
Here's the commands I used:

```
# Log in to the Docker container
docker exec -it outline-postgres-1 psql -U postgres

# Check if the desired user is an admin ("f" = false, "t" = true)
SELECT id,username,"isAdmin" FROM users WHERE username = 'username_goes_here';

# Update the desired user's "isAdmin" flag to TRUE
UPDATE users SET "isAdmin" = TRUE WHERE username = 'username_goes_here';

# Verify the update was made (should now show "t")
SELECT id,username,"isAdmin" FROM users WHERE username = 'username_goes_here';
```

That user should now have admin. It feels weird that this is the official way to set the initial admin, hopefully this process will be improved in the future without having to manually edit the database. For instance, in Gitea the first user to register is an admin. This seems fair since I would hope whomever is setting up an instance would also be trusted to have admin powers.

**Reply by uirmaksever** [reply]
Just wanted to note that this answer is not valid anymore since structure of users table has changed, probably due to later updates. there is no isAdmin column anymore but there is a "role" column, which initially had the value of "member" My solution was changing the "role" column in users table from "member" to "admin", for the user to be converted to admin. After that, I was able to see Workspace admin options on UI.

**Reply by novo-github** [reply]
Commands I was currently able to use with the latest self-hosted version (v0.78.0)

```
# Log in to the Docker container
docker exec -it outline-postgres-1 psql -U postgres -d outline

# Check if the desired user is an admin
SELECT id, name, role FROM users WHERE name = 'your_username';
# This should show the user has a 'member' role

# Update the role of the user 'your_username' to 'admin'
UPDATE users SET role = 'admin' WHERE name = 'your_username';

# Verify the update
SELECT id, name, role FROM users WHERE name = 'your_username';
```

My first user was created with the self-hosted OIDC (Mattermost). Took me a while to get to here. Hope an API would be available soon for this, for easier configuration

**Reply by 3isenHeiM** [reply]
Is there a fix planned to assign a role ID (or priv level) to the SSO users ?
